### PR TITLE
[OpenMP]Support for lowering masked op

### PR DESCRIPTION
**Pull request by anchuraj**
Change adds the lowering support for Omp masked directive.  Omp masked is introduced in 5.2 standard and allows a parallel region to be executed by threads specified by a programmer. This is achieved with the help of filter clause which helps to specify thread id expected to execute the region.

Other related PRs: 
- [Fortran Parsing and Semantic Support](https://github.com/llvm/llvm-project/pull/91432) - Merged
- [MLIR Support](https://github.com/llvm/llvm-project/pull/96022/files) - Merged
- [MaskedOp invocation from flang](https://github.com/llvm/llvm-project/pull/98423) - Merged

### CODE REVIEW

File: mlir/test/Target/LLVMIR/openmp-llvm.mlir [hunk 1]
```diff
@@ -310,6 +310,35 @@ llvm.func @test_omp_master() -> () {
...
+// CHECK: omp.masked.region
+// CHECK: call void @__kmpc_end_masked(ptr @{{[0-9]+}}, i32 [[OMP_THREAD_3_4]])
+// CHECK: br label %omp_region.end
+      omp.terminator
+    }
+    omp.terminator
+  }
+  omp.parallel {
+    omp.parallel {
+      omp.masked filter(%arg0: i32){
```
Comment (skatrak) [inline mlir/test/Target/LLVMIR/openmp-llvm.mlir:330]: Missing CHECK comments here?

**Review by skatrak — APPROVED** [review]
Thank you Anchu, this LGTM. Only a small comment.

File: mlir/test/Target/LLVMIR/openmp-llvm.mlir (hunk 1 above)
Comment (anchuraj) [reply mlir/test/Target/LLVMIR/openmp-llvm.mlir:330]: No, `master` test case had a similar structure where tested code was repeated. I thought it was added for readability purposes. I realise from your question, it adds more of confusion. I will remove the unnecessary code.

**Review by tblah — APPROVED** [review]
LGTM, thanks!

**Review by mjklemm — APPROVED** [review]
LGTM

**Pull request merged**